### PR TITLE
Feat/extend api metadata

### PR DESCRIPTION
**Pull request by dvonrohr**

### CODE REVIEW

File: docs/00-installation-configuration.md [hunk 1]
```diff
@@ -27,6 +27,9 @@ pimcore_monitor:
 
         # Default health report API endpoint to send data to.
         default_endpoint: 'https://health.example.com/report'
+
+        # Environment description for the instance (e.g. "Production", "Staging", "Development"). 
+        instance_environment: 'Production'
```
Comment (aarongerig) [inline docs/00-installation-configuration.md]: I would use lower case and shortened keywords as example values.
```suggestion
        # Environment description for the instance (e.g. "prod", "test", "dev"). 
        instance_environment: 'prod'
```

File: src/PimcoreMonitorBundle/Command/HealthReportCommand.php [hunk 2]
```diff
@@ -76,10 +77,10 @@ protected function configure(): void
      */
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $instanceId = $this->getInstanceId();
+        $projectId = $this->getInstanceId();
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Command/HealthReportCommand.php]: I think we should stick to the old name of "instance ID" nonetheless.
```suggestion
        $instanceId = $this->getInstanceId();
```

File: src/PimcoreMonitorBundle/Command/HealthReportCommand.php [hunk 3]
```diff
@@ -76,10 +77,10 @@ protected function configure(): void
...
         $hostDomain = $this->systemConfig['general']['domain'];
 
-        if (null === $instanceId) {
+        if (null === $projectId) {
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Command/HealthReportCommand.php]:
```suggestion
        if (null === $instanceId) {
```

File: src/PimcoreMonitorBundle/Command/HealthReportCommand.php [hunk 4]
```diff
@@ -105,9 +106,12 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $response = $this->httpClient->request('PUT', $input->getOption('endpoint'), [
                 'auth_bearer' => $this->apiKey,
                 'json' => [
-                    'instance_id' => $instanceId,
-                    'host_domain' => $hostDomain,
+                    'project_id'=> $projectId,
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Command/HealthReportCommand.php]:
```suggestion
                    'instance_id'=> $instanceId,
```

File: src/PimcoreMonitorBundle/DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -52,6 +52,10 @@ private function buildReportNode(): NodeDefinition
                     ->info('Default health report API endpoint to send data to.')
                     ->isRequired()
                 ->end()
+                ->scalarNode('instance_environment')
+                    ->info('Environment for project')
+                    ->isRequired()
+                ->end()
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/DependencyInjection/Configuration.php]: Here I would add a default value of `prod`.

**Review by aarongerig — CHANGES_REQUESTED** [review]
@dvonrohr Some minor changes as discussed.

File: src/PimcoreMonitorBundle/DependencyInjection/Configuration.php [hunk 6]
```diff
@@ -52,6 +52,13 @@ private function buildReportNode(): NodeDefinition
                     ->info('Default health report API endpoint to send data to.')
                     ->isRequired()
                 ->end()
+                ->scalarNode('instance_environment')
+                    ->info('Environment for project')
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/DependencyInjection/Configuration.php]:
```suggestion
                    ->info('The name of the environment in which the application is running in.')
```

File: src/PimcoreMonitorBundle/Command/HealthReportCommand.php [hunk 7]
```diff
@@ -91,6 +92,10 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             return Command::FAILURE;
         }
 
+        if (empty($this->instanceEnvironment)) {
+            $this->instanceEnvironment = 'prod';
+        }
+
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Command/HealthReportCommand.php]: Not necessary, as the property can never be empty at this point.

```suggestion
```

File: src/PimcoreMonitorBundle/Command/HealthReportCommand.php [hunk 8]
```diff
@@ -105,9 +110,12 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             $response = $this->httpClient->request('PUT', $input->getOption('endpoint'), [
                 'auth_bearer' => $this->apiKey,
                 'json' => [
-                    'instance_id' => $instanceId,
-                    'host_domain' => $hostDomain,
+                    'instance_id'=> $instanceId,
```
Comment (aarongerig) [inline src/PimcoreMonitorBundle/Command/HealthReportCommand.php]:
```suggestion
                    'instance_id' => $instanceId,
```

**Review by aarongerig — CHANGES_REQUESTED** [review]

**Review by aarongerig — APPROVED** [review]

**Comment by dvonrohr** [comment]
The changes from this pull request have been successfully integrated into the main branch. Closing this pull request now.

**Pull request closed without merging**